arrow-core-data/src/main/kotlin/arrow/core/NonEmptyList.kt

```diff
@@ -266,6 +279,23 @@ class NonEmptyList<out A>(
   override fun toString(): String =
     show(Show.any())
 
+  fun <B> align(b: NonEmptyList<B>): NonEmptyList<Ior<A, B>> =
+    NonEmptyList(Ior.Both(head, b.head), tail.align(b.tail))
+
+  fun salign(SA: Semigroup<@UnsafeVariance A>, b: NonEmptyList<@UnsafeVariance A>): NonEmptyList<A> =
+    SA.run {
+      NonEmptyList(head.combine(b.head), tail.salign(SA, b.tail).toList())
+    }
+
+  fun <B> padZip(other: NonEmptyList<B>): NonEmptyList<Tuple2<A?, B?>> =
+    NonEmptyList(Tuple2(head, other.head), tail.padZip(other.tail))
+
+  fun <B> zip(other: NonEmptyList<B>): NonEmptyList<Tuple2<A, B>> =
+    NonEmptyList(Tuple2(head, other.head), tail.zip(other.tail).map { Tuple2(it.first, it.second) })
+
+  fun <B, C> zipWith(other: NonEmptyList<B>, f: Function2<A, B, C>): NonEmptyList<C> =
```

Comment:
```suggestion
  fun <B, C> zip(other: NonEmptyList<B>, f: (A, B) -> C): NonEmptyList<C> =
```

This is called `zip` in Kotlin Std.